test/test_models.py

```diff
@@ -220,399 +218,389 @@ def get_export_import_copy(m):
-                break
-        new_model = torch.nn.Sequential(layers)
-        return new_model
-
-    def test_memory_efficient_densenet(self):
-        input_shape = (1, 3, 300, 300)
-        x = torch.rand(input_shape)
-
-        for name in ['densenet121', 'densenet169', 'densenet201', 'densenet161']:
-            model1 = models.__dict__[name](num_classes=50, memory_efficient=True)
-            params = model1.state_dict()
-            num_params = sum([x.numel() for x in model1.parameters()])
-            model1.eval()
-            out1 = model1(x)
-            out1.sum().backward()
-            num_grad = sum([x.grad.numel() for x in model1.parameters() if x.grad is not None])
-
-            model2 = models.__dict__[name](num_classes=50, memory_efficient=False)
-            model2.load_state_dict(params)
-            model2.eval()
-            out2 = model2(x)
-
-            self.assertTrue(num_params == num_grad)
-            torch.testing.assert_close(out1, out2, rtol=0.0, atol=1e-5)
-
-    def test_resnet_dilation(self):
-        # TODO improve tests to also check that each layer has the right dimensionality
-        for i in product([False, True], [False, True], [False, True]):
-            model = models.__dict__["resnet50"](replace_stride_with_dilation=i)
-            model = self._make_sliced_model(model, stop_layer="layer4")
-            model.eval()
-            x = torch.rand(1, 3, 224, 224)
-            out = model(x)
-            f = 2 ** sum(i)
-            self.assertEqual(out.shape, (1, 2048, 7 * f, 7 * f))
+def _make_sliced_model(model, stop_layer):
+    layers = OrderedDict()
+    for name, layer in model.named_children():
+        layers[name] = layer
+        if name == stop_layer:
+            break
+    new_model = torch.nn.Sequential(layers)
+    return new_model
+
+
+def test_memory_efficient_densenet():
+    input_shape = (1, 3, 300, 300)
+    x = torch.rand(input_shape)
+
+    for name in ['densenet121', 'densenet169', 'densenet201', 'densenet161']:
+        model1 = models.__dict__[name](num_classes=50, memory_efficient=True)
+        params = model1.state_dict()
+        num_params = sum([x.numel() for x in model1.parameters()])
+        model1.eval()
+        out1 = model1(x)
+        out1.sum().backward()
+        num_grad = sum([x.grad.numel() for x in model1.parameters() if x.grad is not None])
+
+        model2 = models.__dict__[name](num_classes=50, memory_efficient=False)
+        model2.load_state_dict(params)
+        model2.eval()
+        out2 = model2(x)
 
-    def test_mobilenet_v2_residual_setting(self):
-        model = models.__dict__["mobilenet_v2"](inverted_residual_setting=[[1, 16, 1, 1], [6, 24, 2, 2]])
+        assert num_params == num_grad
+        torch.testing.assert_close(out1, out2, rtol=0.0, atol=1e-5)
+
+
+def test_resnet_dilation():
+    # TODO improve tests to also check that each layer has the right dimensionality
+    for i in product([False, True], [False, True], [False, True]):
```

Comment:
here we should have a parametrization as well, something like:

```py
pytest.mark.parametrize('diltate_layer_2', (True, False))
pytest.mark.parametrize('diltate_layer_3', (True, False))
pytest.mark.parametrize('diltate_layer_4', (True, False))
def test_resnet_dilation(diltate_layer_2, diltate_layer_3, diltate_layer_4):
```
and replace `i` by `(diltate_layer_2, diltate_layer_3, diltate_layer_4)`